destination/src/main/java/com/userleap/destination/SprigDestination.kt

```diff
@@ -0,0 +1,111 @@
+package com.userleap.destination
+
+import android.app.Activity
+import android.app.Application
+import androidx.annotation.VisibleForTesting
+import androidx.fragment.app.FragmentActivity
+import com.segment.analytics.kotlin.android.plugins.AndroidLifecycle
+import com.segment.analytics.kotlin.core.*
+import com.segment.analytics.kotlin.core.platform.DestinationPlugin
+import com.segment.analytics.kotlin.core.platform.Plugin
+import com.userleap.Sprig
+import com.userleap.SurveyState
+import com.userleap.destination.SprigDestination.Companion.EMAIL_KEY
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import java.lang.ref.WeakReference
+
+@Serializable
+data class SprigSettings(val envId: String)
+
+class SprigDestination(
+    private val application: Application
+) : DestinationPlugin(), AndroidLifecycle {
+
+    override val key: String = KEY
+
+    @VisibleForTesting
+    var sprigSettings: SprigSettings? = null
+
+    // The current active activity to show any surveys on
+    private var activityReference: WeakReference<FragmentActivity>? = null
+
+    override fun update(settings: Settings, type: Plugin.UpdateType) {
+        super.update(settings, type)
+        sprigSettings = settings.destinationSettings<SprigSettings>(key)?.also {
+            it.envId.ifNotEmpty()?.let { environment ->
+                Sprig.configure(application, environment)
+            }
+        }
+    }
+
+    override fun identify(payload: IdentifyEvent) = payload.also {
+        recordAnonymousId(it)
+        Sprig.setUserIdentifier(it.userId)
```

Comment:
we should probably use the `identifyAndSetAttributes` method here as well. But that requires an android sdk update first.

Reply:
I'll make the update on iOS first and then you can refer to my change to update android